OpenDreamRuntime/Procs/Native/DreamProcNativeIcon.cs

```diff
@@ -17,6 +18,41 @@ static class DreamProcNativeIcon {
             return new DreamValue(dreamIconObject.Description.Height);
         }
 
+        [DreamProc("Insert")]
+        [DreamProcParameter("new_icon", Type = DreamValue.DreamValueType.DreamObject)]
+        [DreamProcParameter("icon_state", Type = DreamValue.DreamValueType.String)]
+        [DreamProcParameter("dir", Type = DreamValue.DreamValueType.Float)]
+        [DreamProcParameter("frame", Type = DreamValue.DreamValueType.Float)]
+        [DreamProcParameter("moving", Type = DreamValue.DreamValueType.Float)]
+        [DreamProcParameter("delay", Type = DreamValue.DreamValueType.Float)]
+        public static DreamValue NativeProc_Insert(DreamObject instance, DreamObject usr, DreamProcArguments arguments) {
+            //TODO Figure out what happens when you pass the wrong types as args
 
+            arguments.GetArgument(0, "new_icon").TryGetValueAsDreamObject(out var new_icon);
+            arguments.GetArgument(1, "icon_state").TryGetValueAsString(out var icon_state);
+            AtomDirection? dir = null;
+            if (arguments.GetArgument(2, "dir").TryGetValueAsInteger(out var dirNum))
+            {
+                dir = (AtomDirection)dirNum;
+            }
+            int? frame = null;
+            if (arguments.GetArgument(3, "frame").TryGetValueAsInteger(out var frameNum))
+            {
+                frame = frameNum;
+            }
+            bool moving = !(arguments.GetArgument(4, "moving").TryGetValueAsInteger(out var movingNum) && movingNum == 0);
+            int? delay = null;
+            if (arguments.GetArgument(5, "delay").TryGetValueAsInteger(out var delayNum))
+            {
+                delay = delayNum;
+            }
+            
+            DreamMetaObjectIcon.DreamIconObject instanceIconObject = DreamMetaObjectIcon.ObjectToDreamIcon[instance];
+            DreamMetaObjectIcon.DreamIconObject newIconObject = DreamMetaObjectIcon.ObjectToDreamIcon[new_icon];
+            instanceIconObject.Description.InsertIcon(newIconObject.Description, icon_state, dir, frame, delay);
+
+            instanceIconObject.Moving = moving ? DreamMetaObjectIcon.DreamIconMovingMode.Movement : DreamMetaObjectIcon.DreamIconMovingMode.NonMovement;
+            return new DreamValue(instanceIconObject);
```

Comment:
This proc doesn't return anything
```suggestion
            return DreamValue.Null;
```